### PR TITLE
[OpenCL] Cleans control_flow_ops.cc file

### PR DESCRIPTION
**Pull request by lukeiwanski**
Mainly re-order code to follow general rule:
- CPU Operation registration
- GPU Operation registration
- SYCL Operation registration

### CODE REVIEW

**Comment by gunan** [comment]
is there only reordering in this PR?
I think we can skip the reordering for now.
We can close the PR if there are no actual code changes.

**Comment by lukeiwanski** [comment]
Reverted ordering in 4691140
There are other changes in the file ( eg registration for Merge and RefMerge )

File: tensorflow/core/kernels/control_flow_ops.cc [hunk 1]
```diff
@@ -112,28 +112,56 @@ REGISTER_GPU_HOST_REF_KERNEL(string);
...
 
 #define REGISTER_SYCL_REF_SWITCH(type)                     \
   REGISTER_KERNEL_BUILDER(Name("RefSwitch")                \
                               .Device(DEVICE_SYCL)         \
                               .HostMemory("pred")          \
                               .TypeConstraint<type>("T"),  \
                           SwitchOp)
-REGISTER_SYCL_REF_SWITCH(bool);
-TF_CALL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_REF_SWITCH);
+TF_CALL_REAL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_REF_SWITCH);
```
Comment (gunan) [inline tensorflow/core/kernels/control_flow_ops.cc]: This is removing registration of RefSwitch for complex number types. Is this intended? (Not mentioned in PR description).

File: tensorflow/core/kernels/control_flow_ops.cc [hunk 2]
```diff
@@ -112,28 +112,56 @@ REGISTER_GPU_HOST_REF_KERNEL(string);
...
   REGISTER_KERNEL_BUILDER(Name("Switch")                 \
                               .Device(DEVICE_SYCL)       \
-                              .TypeConstraint<type>("T") \
-                              .HostMemory("pred"),       \
+                              .HostMemory("pred")        \
+                              .TypeConstraint<type>("T"),\
                           SwitchOp)
-REGISTER_SYCL_KERNEL(bool);
-TF_CALL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_KERNEL);
+TF_CALL_REAL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_SWITCH);
```
Comment (gunan) [inline tensorflow/core/kernels/control_flow_ops.cc]: This is removing registration of Switch for complex number types. Is this intended? (Not mentioned in PR description).

File: tensorflow/core/kernels/control_flow_ops.cc [hunk 3]
```diff
@@ -399,25 +452,16 @@ REGISTER_GPU_KERNEL(bool);
...
-  Name("Exit").Device(DEVICE_SYCL).TypeConstraint<type>("T"), ExitOp)
+      Name("Exit").Device(DEVICE_SYCL).TypeConstraint<type>("T"), ExitOp);
 REGISTER_SYCL_KERNEL(bool);
 TF_CALL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_KERNEL);
 
-#define REGISTER_SYCL_REF_KERNEL(type)  \
-  REGISTER_KERNEL_BUILDER(              \
-  Name("RefExit").Device(DEVICE_SYCL).TypeConstraint<type>("T"), ExitOp)
-REGISTER_SYCL_REF_KERNEL(bool);
-TF_CALL_NUMBER_TYPES_NO_INT32(REGISTER_SYCL_REF_KERNEL);
```
Comment (gunan) [inline tensorflow/core/kernels/control_flow_ops.cc]: Removing registration of RefExit intended? Maybe you wanted to pack this macro into the REGISTER_SYCL_KERNEL macro above?

File: tensorflow/core/kernels/control_flow_ops.cc (hunk 2 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/control_flow_ops.cc]: yes that was intended as complex numbers are not implemented in Eigen ( yet )

File: tensorflow/core/kernels/control_flow_ops.cc (hunk 1 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/control_flow_ops.cc]: yes that was intended as complex numbers are not implemented in Eigen ( yet )

File: tensorflow/core/kernels/control_flow_ops.cc (hunk 3 above)
Comment (lukeiwanski) [inline tensorflow/core/kernels/control_flow_ops.cc]: Is it meant to be registered for GPU? If so, GPU registration is missing too
 https://github.com/lukeiwanski/tensorflow/blob/469114017ba41b3447248c074d448167b7c6aee1/tensorflow/core/kernels/control_flow_ops.cc#L446

Comment (gunan) [inline tensorflow/core/kernels/control_flow_ops.cc]: I think that was a mistake which is being rectified right now with another PR.

Comment (lukeiwanski) [inline tensorflow/core/kernels/control_flow_ops.cc]: RefExit is registered for the same types as Exit in c607854

**Comment by gunan** [comment]
Jenkins, test this please.

**Review by gunan — APPROVED** [review]

**Pull request merged**